x-pack/plugins/maps/public/classes/styles/vector/components/data_mapping/percentiles_form.tsx

```diff
@@ -0,0 +1,148 @@
+        if (typeof percentile === 'number') {
+          let delta = 1;
+          if (index === this.state.percentiles.length - 1) {
+            // Adding row to end of list.
+            if (index !== 0) {
+              const prevPercentile = this.state.percentiles[index - 1];
+              if (typeof prevPercentile === 'number') {
+                delta = percentile - prevPercentile;
+              }
+            }
+          } else {
+            // Adding row in middle of list.
+            const nextPercentile = this.state.percentiles[index + 1];
+            if (typeof nextPercentile === 'number') {
+              delta = (nextPercentile - percentile) / 2;
+            }
+          }
+          newPercentile = percentile + delta;
+          if (newPercentile >= 100) {
+            newPercentile = 99;
+          }
+        }
+
+        const percentiles = [
+          ...this.state.percentiles.slice(0, index + 1),
+          newPercentile,
+          ...this.state.percentiles.slice(index + 1),
+        ];
+        this.setState({ percentiles });
+      };
+
+      const onRemove = () => {
+        const percentiles =
+          this.state.percentiles.length === 1
+            ? this.state.percentiles
+            : [
+                ...this.state.percentiles.slice(0, index),
+                ...this.state.percentiles.slice(index + 1),
+              ];
+        this.setState({ percentiles });
+      };
+
+      const onPercentileChange = (event: ChangeEvent<HTMLInputElement>) => {
+        const sanitizedValue = parseFloat(event.target.value);
+        const percentiles = [...this.state.percentiles];
+        percentiles[index] = isNaN(sanitizedValue) ? '' : sanitizedValue;
+        this.setState({ percentiles });
+      };
+
+      const isInvalid = isInvalidPercentile(percentile);
+      const error = isInvalid
+        ? i18n.translate('xpack.maps.styles.invalidPercentileMsg', {
+            defaultMessage: `Percentile must be a number between 0 and 100, exclusive`,
+          })
+        : null;
+
+      return (
+        <EuiFormRow key={index} display="rowCompressed" isInvalid={isInvalid} error={error}>
+          <EuiFieldNumber
+            isInvalid={isInvalid}
+            value={percentile}
+            onChange={onPercentileChange}
+            append={
+              <RowActionButtons
+                onAdd={onAdd}
+                onRemove={onRemove}
+                showDeleteButton={this.state.percentiles.length > 1}
+              />
+            }
+            compressed
+          />
+        </EuiFormRow>
+      );
+    });
+
+    const applyButton = !_.isEqual(this.state.percentiles, this.props.initialPercentiles) ? (
+      <>
+        <EuiSpacer />
+        <EuiTextAlign textAlign="right">
+          <EuiButton
+            fill
+            isDisabled={this.state.percentiles.some(isInvalidPercentile)}
+            onClick={this._onSubmit}
+            size="s"
+          >
+            <FormattedMessage
```

Comment:
consider removing and have `this._onSubmit` fire automagically. (providing inputs are valid). It might save a click. (this is kind of nice in Maps imho, that so much of the interactions are instantaneous).